test/python/pulse/test_builder.py

```diff
@@ -0,0 +1,969 @@
+                    pulse.delay(5, d1)
+                    pulse.delay(7, d0)
+
+        schedule = transforms.remove_directives(schedule)
+
+        reference = pulse.Schedule()
+        # d0
+        reference = reference.insert(0, instructions.Delay(3, d0))
+        reference = reference.insert(7, instructions.Delay(7, d0))
+        # d1
+        reference = reference.insert(9, instructions.Delay(5, d1))
+        # d2
+        reference = reference.insert(3, instructions.Delay(11, d2))
+
+        self.assertEqual(schedule, reference)
+
+    def test_barrier_with_align_left(self):
+        """Test barrier directive with left alignment context."""
+        d0 = pulse.DriveChannel(0)
+        d1 = pulse.DriveChannel(1)
+        d2 = pulse.DriveChannel(2)
+
+        with pulse.build() as schedule:
+            with pulse.align_left():
+                pulse.delay(3, d0)
+                pulse.barrier(d0, d1, d2)
+                pulse.delay(11, d2)
+                with pulse.align_left():
+                    pulse.delay(5, d1)
+                    pulse.delay(7, d0)
+
+        schedule = transforms.remove_directives(schedule)
+
+        reference = pulse.Schedule()
+        # d0
+        reference += instructions.Delay(3, d0)
+        reference += instructions.Delay(7, d0)
+        # d1
+        reference = reference.insert(3, instructions.Delay(5, d1))
+        # d2
+        reference = reference.insert(3, instructions.Delay(11, d2))
+
+        self.assertEqual(schedule, reference)
+
+    def test_barrier_on_qubits(self):
+        """Test barrier directive on qubits."""
+        with pulse.build(self.backend) as schedule:
+            pulse.barrier(0, 1)
+
+        reference = pulse.Schedule()
+        reference += directives.RelativeBarrier(
+            pulse.DriveChannel(0),
+            pulse.DriveChannel(1),
+            pulse.MeasureChannel(0),
+            pulse.MeasureChannel(1),
+            pulse.ControlChannel(0),
+            pulse.ControlChannel(1),
+            pulse.AcquireChannel(0),
+            pulse.AcquireChannel(1)
+        )
+        self.assertEqual(schedule, reference)
+
+    def test_trivial_barrier(self):
+        """Test that trivial barrier is not added."""
+        with pulse.build() as schedule:
+            pulse.barrier(pulse.DriveChannel(0))
+
+        self.assertEqual(schedule, pulse.Schedule())
+
+
+class TestUtilities(TestBuilder):
+    """Test builder utilities."""
+
+    def test_active_backend(self):
+        """Test getting active builder backend."""
+        with pulse.build(self.backend):
+            self.assertEqual(pulse.active_backend(), self.backend)
+
+    def test_append_block(self):
+        """Test appending a block to the active builder."""
+        d0 = pulse.DriveChannel(0)
+        block = pulse.Schedule()
+        block += instructions.Delay(10, d0)
+
+        schedule = pulse.Schedule()
+        with pulse.build() as schedule:
+            builder.append_block(block)
```

Comment:
same as `pulse.call(block)`?

Reply:
Currently, it is, but this is just due to the current implementation. I expect this to change very soon.